qt/draw_widget.cpp

```diff
@@ -214,32 +214,59 @@ bool DrawWidget::event(QEvent * event)
 #if !defined(OMIM_OS_LINUX)
     return QOpenGLWidget::event(event);
 #else
-  auto dfTouchEventType = qtTouchEventTypeToDfTouchEventType(event->type());
-  if (dfTouchEventType == df::TouchEvent::TOUCH_NONE)
-    return QOpenGLWidget::event(event);
-
-  event->accept();
-  QTouchEvent const * qtTouchEvent = dynamic_cast<QTouchEvent const *>(event);
-  df::TouchEvent dfTouchEvent;
-  // The SetTouchType hast to be set even if `qtTouchEvent->points()` is empty
-  // which theoretically can happen in case of `QEvent::TouchCancel`
-  dfTouchEvent.SetTouchType(dfTouchEventType);
-
-  int64_t i = 0;
-  for (auto it = qtTouchEvent->points().cbegin();
-       it != qtTouchEvent->points().cend() && i < 2; /* For now drape_frontend can only handle max 2 touches */
-       ++it, ++i)
+  // TouchScreen
+  if (auto dfTouchEventType = qtTouchEventTypeToDfTouchEventType(event->type());
```

Comment:
Why is this code Linux-only? Are there any cross-platform Qt implementations we can reuse everywhere?

Reply:
Are you asking me why a bunch of things like the [QNativeGestureEvent behaviour is documented to be platform dependent](https://doc.qt.io/qt-6/qnativegestureevent.html)? Next to another bunch of behaviors, that are not documented but de facto platform dependent (meaning you can try it out yourself, plus the forums are full of posts confirming that).
If that is the case, then I am not the right person to answer that.

Reply:
Why can't this code be used on all platforms? E.g. on mac and linux at the same time? It's more about ifdef necessity.

Reply:
Context:
https://github.com/organicmaps/organicmaps/pull/9616
https://github.com/organicmaps/organicmaps/pull/9635